types/index.d.ts

```diff
@@ -1,18 +1,27 @@
 
 declare module 'replace-in-file' {
-  function replaceInFile(config: ReplaceInFileConfig): Promise<ReplaceResult[]>;
-  function replaceInFile(config: ReplaceInFileConfig, cb: (error: Error, results: ReplaceResult[]) => void): void;
+  export function replaceInFile(config: ReplaceInFileConfig): Promise<ReplaceResult[]>;
+  export function replaceInFile(config: ReplaceInFileConfig, cb: (error: Error, results: ReplaceResult[]) => void): void;
   export default replaceInFile;
 
   namespace replaceInFile {
     export function sync(config: ReplaceInFileConfig): ReplaceResult[];
+    export function replaceInFileSync(config: ReplaceInFileConfig): ReplaceResult[];
+    export function replaceInFile(config: ReplaceInFileConfig): Promise<ReplaceResult[]>;
+    export function replaceInFile(config: ReplaceInFileConfig, cb: (error: Error, results: ReplaceResult[]) => void): void;
   }
 
+  export function sync(config: ReplaceInFileConfig): ReplaceResult[];
+  export function replaceInFileSync(config: ReplaceInFileConfig): ReplaceResult[];
+
+  export type From = string | RegExp | FromCallback;
+  export type To = string | ToCallback;
+
   export interface ReplaceInFileConfig {
     files: string | string[];
     ignore?: string | string[];
-    from: string | RegExp | string[] | RegExp[] | FromCallback;
-    to: string | string[] | ToCallback;
+    from: From | Array<From>;
+    to: To | Array<To>;
```

Comment:
Would this change not also suggest an Array of `FromCallback` type could happen?

Reply:
It's possible, but let's inspect the current implementation:
1) `replaceInFileSync` invokes `parseConfig`
2) `parseConfig` optionally handles `files` and `ignore` fields, injects `defaultConfig` and returns it back
```
Object.assign({}, defaults, config)
```
3) Then config params are passed to `replaceSync` 
```
const {files, from, to, dry, verbose} = config;
...
paths.map(path => replaceSync(path, from, to, config));
```
4) and then to `makeReplacements` 
```
makeReplacements(
    contents, from, to, file, countMatches
  );
```
5) where `from` turns into array
```
  if (!Array.isArray(from)) {
    from = [from];
  }
```
6) which is reduced
```
const newContents = from.reduce((contents, item, i) => {
  // ...
    if (typeof item === 'function') {
      item = item(file);
    }
  // ...
}
```
It looks that `Array<FromCallback>` is supported.